### ISSUE TITLE
Does Keras disable regularization (dropout, noise) layers when evaluating the validation data during Model.fit()?

### ISSUE THREAD

**Issue by mcourteaux** [tensorflow/tensorflow#37792]
## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit
https://www.tensorflow.org/api_docs/python/tf/keras/Model#predict

### Clear description

I can't find a clear statement whether or not the regularization layers (noise, dropout) are bypassed when the validation data is processed (to calculate the validation loss) when calling `Model.fit()` provided with validation data. I can see that in the source code of the Dropout Layer that it is branched based on the `training` argument of `call()`. https://github.com/tensorflow/tensorflow/blob/v2.1.0/tensorflow/python/keras/layers/core.py#L181

But it is completely unclear to me whether or not the validation pass is considered "training" or not. After all, I'm calling the "fit" function.

**Comment by khimraj** [comment]
@mcourteaux, first of all validation pass is not considered as training and during training of network actually dropout layer is not bypassed instead of that all neuron's weight of dropout layer is multiplied by probability p of its existence. 
![Screenshot from 2020-03-22 09-45-00](https://user-images.githubusercontent.com/36735394/77242230-5c36bb00-6c22-11ea-9633-92394b501149.png)
For more detail you can refer [this](http://www.jmlr.org/papers/volume15/srivastava14a/srivastava14a.pdf) paper.

**Comment by mcourteaux** [comment]
So, based on your statement "validation pass is not considered as training", all types of noise and dropout do not affect validation loss computations.

**Comment by khimraj** [comment]
Yes validation pass is not considered as training and dropout layer doesn't affect validation loss.

**Comment by riemanli** [comment]
> Yes validation pass is not considered as training and dropout layer doesn't affect validation loss.

Hi. I can clearly see that dropout is not considered during validation from your answer. But I was wondering if this is also true for regularization. Thank you.

**Comment by mcourteaux** [comment]
Me now being a little more familiar with the code, I would say: Yes, it will be considered. The loss function never changes, not in training or validation, or evaluation. Just the noise/dropout layer is bypassed, and the network produces a prediction in the fullest "quality" (ie: without the distortions from those layers).
